ipywidgets/widgets/widget.py

```diff
@@ -223,6 +272,10 @@ def open(self):
             if buffers:
                 # FIXME: workaround ipykernel missing binary message support in open-on-init
                 # send state with binary elements as second message
+                # TODO: if this gets fixed, _split_state_buffers does not need to have a seperate
```

Comment:
(and a few other places, basically we keep the serialized state as binary until *just* before we send it, and we deserialize the binary state *just* after we get the message.)